python/graphstorm/model/node_glem.py

```diff
@@ -125,8 +143,23 @@ def set_gnn_encoder(self, gnn_encoder):
 
     @property
     def gnn_encoder(self):
-        """Alias for accessing the gnn_encoder"""
-        return self.gnn.gnn_encoder
+        """Alias for accessing the gnn_encoder. Hide gnn_encoder if the inference route is lm"""
+        return self.gnn.gnn_encoder if self.inference_route_is_gnn else None
+
+    @property
+    def node_input_encoder(self):
+        """Alias for accessing the node_input_encoder"""
+        return self.lm.node_input_encoder
+
+    @property
+    def _node_input_encoder(self):
```

Comment:
why do we have this property?

Reply:
i don't see this property is used. Can you remove it?

Reply:
removed